ddtrace/debugging/_capture/collector.py

```diff
@@ -0,0 +1,110 @@
+from typing import Any
+from typing import Callable
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from ddtrace.debugging._capture.model import CaptureState
+from ddtrace.debugging._capture.model import CapturedEvent
+from ddtrace.debugging._encoding import BufferedEncoder
+from ddtrace.debugging._metrics import metrics
```

Comment:
nit (just complete when read, not preventing approval): if these are used only for typing (can't find other uses in the PR) they could go inside a:

```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from ddtrace.debugging._encoding import BufferedEncoder
```

This way you avoid some imports at runtime.